### PR TITLE
Added possibility to select slice when converting MultiDimensionalArray to an EigenMatrix

### LINKED ISSUE #88: Allow to choose the indexes of the `MultiDimensionalArray` to map into the Eigen Matrix in the`to_eigen` method
Hi, 
I am working in loading some complex MAT file in C++ using matioCpp and then obtaining their corresponding Eigen Matrix. One of the data structure I need loaded is of type `MultiDimensionalArray` with dimension (3,1,15000). 

Up to now is not possible to directly use the `to_eigen` method https://github.com/ami-iit/matio-cpp/blob/1a846dc8febc93ee7f3b03a2b78336d8d73e765d/include/matioCpp/impl/EigenConversions.tpp#L13-L19
to convert it to an Eigen Matrix since the array's dimensions are > 2. 

It would be nice if I could select the indices of the dimensions I need when calling this method (in this case they are the 1st and 3rd dimension).

### PR DESCRIPTION
**Pull request by S-Dafarra**
Fixes #88

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -153,6 +153,44 @@ Eigen::Vector3i eigenVec;
...
+{
+    tensor[i] = i + 1.0;
+}
+
+matioCpp::MultiDimensionalArray<float> matioCppMatrix2("matrix", { 2, 2, 3 }, tensor.data());
+
+/*
+So we have a tensor of the type
+| 1  3 | | 5 6 | |  9 11 |
+| 2  4 | | 7 8 | | 10 12 |
```
Comment (xela-95) [inline README.md:169]: I still have to run the code, but I think there's a typo here:

```suggestion
| 1  3 | | 5 7 | |  9 11 |
| 2  4 | | 6 8 | | 10 12 |
```

File: README.md [hunk 2]
```diff
@@ -153,6 +153,44 @@ Eigen::Vector3i eigenVec;
...
+Obtain
+| 1  3 |
+| 2  4 |
+*/
+
+Eigen::MatrixXf slice2 = matioCpp::to_eigen(matioCppMatrix2, { 1, -1, -1 }; //Equivalent to the Matlab operation matioCppMatrix2(2,:,:)
+/*
+Obtain
+| 2 7 10|
+| 4 8 12|
```
Comment (xela-95) [inline README.md:183]: See comment above

```suggestion
| 2 6 10|
| 4 8 12|
```

File: test/ExogenousConversionsUnitTest.cpp [hunk 3]
```diff
@@ -85,6 +85,56 @@ TEST_CASE("Eigen Conversions")
...
+        {
+            tensor[i] = i + 1.0;
+        }
+
+        matioCpp::MultiDimensionalArray<float> matioCppMatrix2("matrix", { 2, 2, 3 }, tensor.data());
+
+        /*
+        So we have a tensor of the type
+        | 1  3 | | 5 6 | |  9 11 |
+        | 2  4 | | 7 8 | | 10 12 |
```
Comment (xela-95) [inline test/ExogenousConversionsUnitTest.cpp:104]: Same as above:

```suggestion
        | 1  3 | | 5 7 | |  9 11 |
        | 2  4 | | 6 8 | | 10 12 |
```

**Review by traversaro — APPROVED** [review]
Super cool!

**Review by xela-95 — APPROVED** [review]
Thanks a lot @S-Dafarra !! By a first look it seems good to me! I left some minor comments related to the example used in documentation and test.

**Comment by S-Dafarra** [comment]
Great catch @xela-95, thanks a lot!

**Comment by S-Dafarra** [comment]
Actually, also the test was bugged and I realized I was missing some checks. Fixing them soon

**Review by xela-95 — APPROVED** [review]

**Pull request merged**